lib/Zonemaster/Backend/DB.pm

```diff
@@ -130,6 +133,30 @@ sub process_unfinished_tests {
     }
 }
 
+sub generate_deterministic_hash {
+    my ( $self, $params ) = @_;
+
+    my $profile = Zonemaster::Engine::Profile->effective;
+
+    my %to_encode = ();
+    $to_encode{domain}      = lc $$params{domain}   // "";
+    $to_encode{ds_info}     = $$params{ds_info}     // [];
+    $to_encode{ipv4}        = $$params{ipv4}        // $profile->get( 'net.ipv4' );
+    $to_encode{ipv6}        = $$params{ipv6}        // $profile->get( 'net.ipv6' );
+    $to_encode{nameservers} = $$params{nameservers} // [];
+    $to_encode{priority}    = $$params{priority}    // 10;
+    $to_encode{profile}     = $$params{profile}     // "default";
```

Comment:
The API documentation should be then be updated to state that "default" will be used if unspecified.

Reply:
It looks like the API already specifies the default values for all params.
https://github.com/zonemaster/zonemaster-backend/blob/develop/docs/API.md#params-2

The `ipv4` and `ipv6` need to be updated though because we now accept unset values (and set them accordingly to the config).